src/source.cpp

```diff
@@ -38,6 +38,10 @@ namespace openmc {
 
 namespace model {
 
+typedef Particle::Bank (*sample_t)(uint64_t  &seed);
```

Comment:
From the C++ core guidelines:
- [Prefer `using` over `typedef` for defining aliases](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rt-using)
- [Use C++-style declarator layout](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-ptr)

Also the argument is supposed to be a pointer, not a reference. This happens to work because, well, references typically get turned into pointers under the hood, but there's no guarantee of that.
```suggestion
using sample_t = Particle::Bank (*)(uint64_t* seed);
```